### PR TITLE
Fixes related to reopen_signal

### PR DESCRIPTION
**Pull request by gautam-kotian-sociomantic**
These are fixes for what I described in https://github.com/sociomantic-tsunami/ocean/issues/486#issuecomment-379847521.

Note that I haven't run any tests for these (unable to build ocean at this point).

### CODE REVIEW

File: src/ocean/util/app/ext/ReopenableFilesExt.d [hunk 1]
```diff
@@ -96,7 +96,7 @@ public class ReopenableFilesExt : IApplicationExtension, ISignalExtExtension
 
     public this ( SignalExt signal_ext = null, int reopen_signal = SIGHUP )
     {
-        if (signal_ext)
+        if (signal_ext && reopen_signal)
```
Comment (gavin-norman-sociomantic) [inline src/ocean/util/app/ext/ReopenableFilesExt.d:103]: Would it make sense to also update the doc for `reopen_signal` to describe this usage?

File: src/ocean/util/app/DaemonApp.d [hunk 2]
```diff
@@ -292,6 +292,10 @@ public abstract class DaemonApp : Application,
             Signal to trigger reopening of files which are registered with the
             ReopenableFilesExt. (Typically used for log rotation.)
 
+            If the command to reopen log files is sent via a UNIX socket, then
```
Comment (gavin-norman-sociomantic) [inline src/ocean/util/app/DaemonApp.d]: Suggestion: "If this field is set to 0, no signal handler will be installed for file reopening. This is useful if your app is deployed to use a different means of triggering file reopening (e.g. a UNIX socket command)."

File: src/ocean/util/app/ext/ReopenableFilesExt.d (hunk 1 above)
Comment (gautam-kotian-sociomantic) [reply src/ocean/util/app/ext/ReopenableFilesExt.d:103]: Yes, I suppose so. But I don't know what's the best way of updating the doc. Let me push a fixup commit as a starting point.

File: src/ocean/util/app/DaemonApp.d (hunk 2 above)
Comment (gautam-kotian-sociomantic) [reply src/ocean/util/app/DaemonApp.d]: Sounds good.

**Comment by gautam-kotian-sociomantic** [comment]
Pushed update with:
- `reopen_signal` documentation updated as suggested
- fixup comment added with proposed changes to documentation of ReopenableFilesExt constructor

<details>
<summary>Diff</summary>

```diff
diff --git a/src/ocean/util/app/DaemonApp.d b/src/ocean/util/app/DaemonApp.d
index 5075e2f..cf71cf5 100644
--- a/src/ocean/util/app/DaemonApp.d
+++ b/src/ocean/util/app/DaemonApp.d
@@ -292,9 +292,10 @@ public abstract class DaemonApp : Application,
             Signal to trigger reopening of files which are registered with the
             ReopenableFilesExt. (Typically used for log rotation.)
 
-            If the command to reopen log files is sent via a UNIX socket, then
-            this should be set to zero so that ReopenableFilesExt does not
-            install any signal handler.
+            If this field is set to 0, no signal handler will be installed for
+            file reopening. This is useful if your app is deployed to use a
+            different means of triggering file reopening (e.g. a UNIX socket
+            command).
 
         ***********************************************************************/
 
diff --git a/src/ocean/util/app/ext/ReopenableFilesExt.d b/src/ocean/util/app/ext/ReopenableFilesExt.d
index ada558c..c0d0fb8 100644
--- a/src/ocean/util/app/ext/ReopenableFilesExt.d
+++ b/src/ocean/util/app/ext/ReopenableFilesExt.d
@@ -83,14 +83,18 @@ public class ReopenableFilesExt : IApplicationExtension, ISignalExtExtension
 
     /***************************************************************************
 
-        Constructor. Optionally registers this extension with the signal
-        extension and activates the handling of the specified signal, which will
-        cause the registered files to be reopened.
+        Constructor.
+
+        Optionally registers this extension with the signal extension and
+        activates the handling of the specified signal, which will cause the
+        registered files to be reopened. For this to happen, a non-null
+        `signal_ext` and a non-zero `reopen_signal` must be supplied.
 
         Params:
-            signal_ext = optional SignalExt instance to register with
-            reopen_signal = if signal_ext is non-null, signal to trigger
-                reopening of registered files
+            signal_ext = SignalExt instance to register with (defaults to null)
+            reopen_signal = signal to trigger reopening of registered files
+                (defaults to SIGHUP, and considered only if non-zero and
+                `signal_ext` is non-null)
 
     ***************************************************************************/
 
```

**Review by mihails-strasuns-sociomantic — COMMENTED** [review]
If this is a new feature, needs release notes.

**Comment by mihails-strasuns-sociomantic** [comment]
(added release notes myself as this was last PR planned for ocean release)

**Pull request merged**